src/reports/components/ReportFilter.jsx

```diff
@@ -90,6 +114,14 @@ const ReportFilter = ({
     }
   };
 
+  useEffect(() => {
+    if (locationState.showOnPageLoad) {
+      // Don't treat re-renders as first page load.
+      locationState.showOnPageLoad = false;
```

Comment:
Not so sure about this, it feels like there has to be a better way to accomplish the same thing.

Without this, the replay filter just recloses immediatly when the users wants to adjust the filter.

Reply:
Yes, this all doesn't look clean enough.

Reply:
I moved this into the parent component. I think this is better?

Reply:
I've moved the entire state into the main page component and pass it down into the filter report now. I don't think having the filter contain the logic for that was the best idea.

I hope this addresses your concern